Integration with spm batch

TomMaullin opened the pull request:
This PR integrates the viewer with SPM, thus addressing issue #27 . Full details of how this is done can be found in the readme.md file.

@cmaumet, Would you mind having a look at this PR and giving some feedback on the user interface when you are back please?

cmaumet commented on Data/jsons/pain_01.json:
```diff
@@ -0,0 +1,1554 @@
+{
```
@TomMaullin: could we remove this file from the codebase?

cmaumet commented on README.md:
```diff
@@ -4,47 +4,59 @@ Read and display NIDM-Results packs in the SPM environment.
+<img src="Doc/example_batch.png" width="200"> |  
+:-------------------------:|
+ **Fig. 3.** The NIDM-Results batch window. |
+5. If there are multiple excursion sets stored in the NIDM-Results file the contrast window will open. Select the contrasts you wish to view and click done.
```
Could we have the option to specify the contrast from the original batch rather than interactively opening the contrast manager? This would be similar to the behavior of 'SPM > Stats > Results Report' and would allow to run the NIDM-Results viewer completely in batch mode.

cmaumet commented on lib/changeNIDMtoSPM.m:
```diff
@@ -66,6 +66,24 @@
         contrastWeightMatrix = relevantToExcursion(contrastWeightMatrix, exID, exLabels);
     end
     
+    %Get the StatType and statisticMaps
```
These changes do not seem to be related to the SPM batch integration. Can you shortly comment on why they appear here? E.g. do they address another issue, is it a bug fix, other?

cmaumet commented on lib/changeNIDMtoSPM.m:
```diff
@@ -78,6 +96,19 @@
     [~, name, ext] = fileparts(designMatrixFilename);
     xXtemp.xKXs.X = csvread(fullfile(filepathTemp, [name, ext]));
     
+    %Record the SVD data.
```
Same comment as above. Can you (very) shortly explain why those changes were made?

TomMaullin replied:
If I am completely honest, I am not sure why I made these changes on this branch. I think I must have thought I was on a different branch. They are useful small changes though in that they result in the SPM object generated by changeNIDMtoSPM more closely resembling that generated by normal SPM output. I'd still recommend keeping them!

cmaumet replied:
Okay! That's fine.

TomMaullin commented:
Hi @cmaumet , I have now addressed all feedback. As it has been a while since we last looked at this PR I thought I'd best confirm, are you still happy for me to merge?

cmaumet commented:
@TomMaullin: This looks good to me. Can you check that the tests pass and copy/paste the test results in a comment below? 

Also, note that you latest commits are not associated with your GitHub profile, you might need to reset your email address in the git configurations: cf. https://help.github.com/articles/setting-your-commit-email-address-in-git/. (That's not a big deal for the commits above but it is worth setting it up now for future commits).

TomMaullin commented:
Hi @cmaumet , I can confirm all tests pass. Here are the results:

result = 

  1×32 TestResult array with properties:

    Name
    Passed
    Failed
    Incomplete
    Duration
    Details

Totals:
   32 Passed, 0 Failed, 0 Incomplete.
   212.8268 seconds testing time.

Ah okay I will look into the email I've used.

cmaumet commented:
Excellent! Thanks! I will let you merge :)
